src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -59,6 +59,13 @@ public static function compile(Route $route)
         $staticPrefix = $result['staticPrefix'];
 
         $pathVariables = $result['variables'];
+
+        foreach ($pathVariables as $pathParam) {
+            if ('_fragment' == $pathParam) {
```

Comment:
Our coding standards are that we always use strict comparison unless we explicitly need the type juggling (because type juggling can actually cause totally unexpected behavior if you are not careful).
And given we know that strings are always returned, we know we don't need this type juggling.